doc/man3/OSSL_HPKE_CTX_new.pod

```diff
@@ -84,48 +84,159 @@ is likely required before using these APIs.  HPKE is used by various
 other IETF specifications, including the (draft) TLS Encrypted Client 
 Hello (ECH) specification and others.
 
-HPKE is a highly flexible consstruct requiring supporting many optional
-inputs for the various combinations in which modes and algorithms can
-be used.
-
-Notes on the various operations supported:
-
-- OSSL_HPKE_seal() is used by senders to encrypt a value to a
-recipient (supplying optional additional authenticated data or
-AAD)
+HPKE is a highly flexible consstruct supporting many optional inputs for the
+various combinations in which modes and algorithms can be used.
+
+=head2 Define'd symbols and data strutures
+
+- Symbols are defined (in include/openssl/hpke.h) for each of the
+IANA-allocated codepoints for HPKE modes ,e.g. OSSL_HPKE_MODE_BASE,
+key encapsulation methods (KEMs), e.g. OSSL_HPKE_KEM_ID_X25519, key
+derivation functions (KDFs), e.g. OSSL_HPKE_KDF_ID_SHA256 and
+authenticated encryption with additional data (AEAD) algorithms,
+e.g. OSSL_HPKE_AEAD_ID_AES_GCM_128.
+
+- The combination of a KEM, KDF and AEAD is represented in the
+OSSL_HPKE_SUITE data structure which maps to how HPKE is used
+in various protocols, and in these APIs.
+
+    typedef struct {
+        uint16_t    kem_id;
+        uint16_t    kdf_id;
+        uint16_t    aead_id;
+    } OSSL_HPKE_SUITE;
+
```

Comment:
Maybe the stuff related to ids code go after the struct here...

e.g The following identifiers are used.

The kem identifiers are .., ...,... and ... 
  
= back

Reply:
attempted that (we'll see how well in a while:-)